pkgs/applications/audio/cadence/default.nix

```diff
@@ -65,10 +83,11 @@
     };
   in lib.mapAttrsToList (script: source: ''
     rm -f ${script}
-    makeWrapper ${python3Packages.python.interpreter} ${script} \
-      --set PYTHONPATH "$PYTHONPATH:${outRef}/share/cadence" \
-      ''${qtWrapperArgs[@]} \
-      --add-flags "-O ${source}"
+    makeQtWrapper ${source} ${script} \
+      --prefix PATH : "${lib.makeBinPath [
+        jack_capture # cadence-render
+        pulseaudioFull # cadence, cadence-session-start
+        ]}"
```

Comment:
doesn't this remove all the important PYTHONPATH changes to the wrapper?
Can you also indent properly?

Reply:
python3 seems to be better than python3Packages, then no need for propagatedBuildInputs and such things in the wrapper. I test heavily to catch others bugs. BTW, i've another thing i'd fix, and i've no idea how to do that properly, I need /nix/store/v433wawv8hl23kb2c2qk0lgg86sainc1-a2jmidid-8/share/dbus-1/services/org.gna.home.a2jmidid.service to be activated .
we can complete with a :

  postInstall = ''
    # D-Bus activation
    install -Dm755 ${a2jmidid}/share/dbus-1/services/org.gna.home.a2jmidid.service -t $out/share/dbus-1/services/
  ''
but user need to reboot for the service to be activated by dbus. There is a lot of issue with dbus activation #17943 for example.

Reply:
@worldofpeace The wrapped python interpreter contains hard-coded references to the correct paths. There is no need to set `PYTHONPATH` when using `withPackages`.